core/domain/takeout_service.py

```diff
@@ -66,5 +66,22 @@ def export_data_for_user(user_id):
         final_name = ('_').join([x.lower() for x in split_name]) + '_data'
         exported_data[final_name] = model.export_data(user_id)
 
-    # Combine the data into a single dictionary.
-    return exported_data
+    # Separate out images.
+    image_paths = {
+        ('user_settings_data', 'profile_picture_data_url'):
+            'user_settings_profile_picture.png'
+    }
+    image_files = []
+    for image_path in image_paths:
+        # Iterate to the position indicated by the tuple.
+        data_pointer = exported_data
+        for key in image_path[:-1]:
```

Comment:
All these references to index "-1" are a bit messy. Can you find a more general way to do this that doesn't require the reader having to decode what the significance of -1 is?

Reply:
Done